### ISSUE TITLE
Does not respect XDG_DOCUMENTS_DIR variable

### ISSUE THREAD

**Issue by vonneudeck** [flathub/us.zoom.Zoom#259]
Applications are supposed to use standard folders, e.g. the "Documents" directory, according to the values set in the corresponding variables like `XDG_DOCUMENTS_DIR`. Usually those are defined in the file `$HOME/.config/user-dirs.dir` and the differ e.g. for different languages or because users set their own values for them.

Unfortunately the Zoom Flatpack seems to have `$HOME/Documents` hardcoded somewhere. It should instead use the locally defined dir.

Issue #257 seems connected.

**Comment by gncs** [comment]
Could this [line](https://github.com/flathub/us.zoom.Zoom/blob/c9576b198bcc79de1ff6657117c4730c0708a176/us.zoom.Zoom.json#L17) be relevant here?

**Comment by danilown** [comment]
Exactly, just submitted a pull request #262

**Comment by danilown** [comment]
Actually, in my humble opinion, I think the correct setting should be `"--filesystem=home`.
I would consider that the correct permission for an app like Zoom and would fix both issues.

**Comment by gncs** [comment]
Ideally, Zoom would store its settings (and recordings, by default) somewhere in `~/.var/app/us.zoom.Zoom`. Further, it would let the user decide where to save the recordings. In that case, Zoom could use this "portal" to obtain permission to save something in a selected location (is this how it works?).

**Comment by danilown** [comment]
For the settings, yes, but to store recordings or other files, I think the default behavior should be to direct the user to his/her `home` folder. 

Currently, Zoom already opens a portal when you try to save a recording, but that seems to be automatically done by GTK [here](https://docs.flatpak.org/en/latest/portals-gtk.html). 

After some investigation, I have found that the problem of not being able to save, for example, a recording on a different folder when using Flatpak might be a bug (or limitation) in the Portal part, because it opens a file selection dialog (it **only** lets you select files) when it should rather be a folder picker or something similar (I'm not a UI guy). If you try to change the default folder for recordings in the settings, the same **file selection** dialog appears, and because we need to press either `cancel` or select a file, Zoom will simply fall back to its default folder `~/Documents/Zoom` and because we are not allowed to change this default path manually (by simply typing it), so we are stuck.

So, to summarise, I'm afraid that this issue cannot be fixed here, and changing the permissions to "--filesystem=home" would not help by any means.

**Comment by gncs** [comment]
Thank you for investigating. Yes, I can confirm that this file/directory chooser is not quite right. It won't let you choose a directory. I thought this was related to the fact that the application doesn't have the right permissions.

Where is the bug of selecting a directory exactly? In Zoom, flatpak, or the way Zoom is wrapped in a flatpak?

**Comment by danilown** [comment]
The problem seems to be the interaction between Zoom (or the GTK version used to build it) and the Portals that is not done correctly. There is nothing that can be changed (at least from my perspective) in the way Zoom is wrapped around Flatpak that can fix this problem.

**Comment by gncs** [comment]
I see. In that case, we should forward a concise description of the problem to the Zoom developers. Maybe we can start a discussion [here](https://devforum.zoom.us/)?

**Comment by muelli** [comment]
> Unfortunately the Zoom Flatpack seems to have `$HOME/Documents` hardcoded somewhere. It should instead use the locally defined dir.

I remember something about Zoom hardcoding that path itself. Is that not true (anymore)?

**Comment by RalfJung** [comment]
It is quite annoying that each time I start zoom it creates this `Documents` folder in `$HOME`, which I don't want or need. So I would much appreciate if that could be fixed.

 I am not even using recordings, so even creating it in the XDG location would still mess up my folder structure.

**Comment by RalfJung** [comment]
> Actually, in my humble opinion, I think the correct setting should be "--filesystem=home".
I would consider that the correct permission for an app like Zoom and would fix both issues.

Uh no please don't. I am relying on the flatpak sandbox to keep Zoom away from my secret keys.

**Comment by Moilleadoir** [comment]
It’s sad that this is still a problem.  Did anyone raise the problem with the developers?  Must admit I don’t have a lot of faith in them.
